python/smashbox/config.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os.path
+import pickle
+from pathlib import Path
+from typing import Any, Literal
+
+import yaml
+from pydantic import BaseSettings
+
+
+logger = None
+
+# this should probably be moved into a utilities module
+def get_logger(name: str = "config", level: int | None = None) -> logging.Logger:
+    global logger
+    if not logger:
+      if level is None:
+          level = logging.INFO  # change here to DEBUG if you want to debug config stuff
+      logging.basicConfig(level=level)
```

Comment:
You can set the logger's log level here if you want to, but the global call should be moved to the entry script.

I know this can be confusing. In Python, loggers can have levels. All messages lower than the logger's level will be suppressed. If the log message is not suppressed by the logger, the global configuration is then checked, if I remember this correctly. If the log message passes this check, it should be displayed.